### PR TITLE
Feature/Webpack Upgrade [OSF-8366]

### PR DESCRIPTION
**Pull request by pattisdr**
## Purpose
We are using a webpack version (1.3.0) that is several major versions out of date. Update the version and migrate our configuration.

## Changes

Upgraded webpack 1.7.2 to webpack 3.8.1 in package.json and admin.package.json, [ and migrating changes ](https://webpack.js.org/guides/migrating/). Upgraded webpack common, admin, prod, karma configs.

## Side effects

Had to upgrade font-awesome-webpack

## Ticket

https://openscience.atlassian.net/browse/OSF-8366

### CODE REVIEW

File: karma.common.conf.js [hunk 1]
```diff
@@ -28,16 +24,16 @@ var webpackTestConfig = {
...
             {test: /\.test\.js$/, loader: 'babel-loader'},
         ])
     },
-        node: {
+    node: {
        fs: 'empty'
     }
 };
-
+webpackTestConfig.resolve.descriptionFiles = ["package.json", "bower.json"],
```
Comment (sloria) [inline karma.common.conf.js]: Drive-by comment: You should be able to use `Object.assign` to merge `webpackCommon.resolve` with more options.

```js
resolve: Object.assign({}, webpackCommon.resolve, {
    descriptionFiles: ['package.json', 'bower.json'],
})
```

Also, as per COSDev, use single-quotes for strings.

Comment (pattisdr) [reply karma.common.conf.js]: ah, thanks

**Comment by sloria** [comment]
I'll take care of resolving the conflict.

**Comment by sloria** [comment]
Updated with `develop` and merged this into `release/next-next-djangole`.
